storm-core/src/clj/org/apache/storm/daemon/executor.clj

```diff
@@ -75,7 +76,7 @@
   "Returns a function that returns a vector of which task indices to send tuple to, or just a single task index."
   [^WorkerTopologyContext context component-id stream-id ^Fields out-fields thrift-grouping ^List target-tasks topo-conf]
   (let [num-tasks (count target-tasks)
-        random (Random.)
+        random (XORShiftRandom.)
```

Comment:
Again here the grouping could be called from multiple threads, but in this case doubling up on some numbers should not be a big deal.